nixos/modules/services/network-filesystems/glusterfs.nix

```diff
@@ -5,6 +5,22 @@ with lib;
 let
   inherit (pkgs) glusterfs rsync;
 
+  tlsCmd = if cfg.enableTLS then
+  ''
+    mkdir -p /var/lib/glusterd
+    touch /var/lib/glusterd/secure-access
+  ''
+  else
+  ''
+    rm -f /var/lib/glusterd/secure-access
```

Comment:
I think you misunderstood me: I think this PR is good, and that deleting `secure-access` here is OK. You just need to document that behaviour in the option `description`.

Also, having the pubkey in the store is OK, it's just the private key that should be a file path instead.